File: ios/XiangshangYouth/Core/Services/MessageApi.swift
import Foundation
struct MessageApi {
    let client: ApiClient
    init(client: ApiClient = .shared) { self.client = client }
    func messages(userID: String) async throws -> [MessageItem] {
        try await client.request(path: "v1/users/\(userID)/messages", type: [MessageItem].self)
    }
    func markRead(messageID: String) async throws {
        try await client.send(path: "v1/messages/\(messageID)/read")
    }
}

File: android/app/src/main/java/com/xiangshang/youth/core/service/MessageApi.kt
package com.xiangshang.youth.core.service

import com.xiangshang.youth.core.model.MessageItem
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

interface MessageApi {
    @GET("v1/users/{userId}/messages")
    suspend fun messages(@Path("userId") userId: String): ApiEnvelope<List<MessageItem>>

    @POST("v1/messages/{messageId}/read")
    suspend fun markRead(@Path("messageId") messageId: String): ApiEnvelope<WriteAck>
}
